reverts https://github.com/mesonbuild/meson/pull/12683
this change leaves no good option since both b_ndebug=true and =false pass
something that may be unwanted
--
--- a/mesonbuild/compilers/cpp.py
+++ b/mesonbuild/compilers/cpp.py
@@ -306,21 +306,7 @@
             return libs
         return []
 
-    def get_assert_args(self, disable: bool) -> T.List[str]:
-        args: T.List[str] = []
-        if disable:
-            return ['-DNDEBUG']
 
-        # Clang supports both libstdc++ and libc++
-        args.append('-D_GLIBCXX_ASSERTIONS=1')
-        if version_compare(self.version, '>=18'):
-            args.append('-D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE')
-        elif version_compare(self.version, '>=15'):
-            args.append('-D_LIBCPP_ENABLE_ASSERTIONS=1')
-
-        return args
-
-
 class ArmLtdClangCPPCompiler(ClangCPPCompiler):
 
     id = 'armltdclang'
@@ -488,17 +474,9 @@
                 assert isinstance(l, str)
             return libs
         return []
-
-    def get_assert_args(self, disable: bool) -> T.List[str]:
-        if disable:
-            return ['-DNDEBUG']
-
-        # XXX: This needs updating if/when GCC starts to support libc++.
-        # It currently only does so via an experimental configure arg.
-        return ['-D_GLIBCXX_ASSERTIONS=1']
 
     def get_pch_use_args(self, pch_dir: str, header: str) -> T.List[str]:
         return ['-fpch-preprocess', '-include', os.path.basename(header)]
